File: diccionariosColecciones/Ejercicio11.py
'''
Realiza un programa que escoja al azar 5 palabras en español del minidiccionario
del ejercicio anterior. El programa irá pidiendo que el usuario
teclee la traducción al inglés de cada una de las palabras y comprobará si
son correctas. Al final, el programa deberá mostrar cuántas respuestas son
válidas y cuántas erróneas.

@author Rafael Infante
'''

from random import randint

'''
variables
'''
cont1 = 0  # contador aciertos
cont2 = 0  # contador fallos

'''
creamos un diccionario vacio
'''
diccionario = {}

'''
creo una lista "es una copia del diccionario, guardo solo claves"
'''
lista = []

'''
creo una lista donde guardo 5 claves
'''
lista5 = []

'''
insertamos claves y valores al diccionario
'''
diccionario['agua'] = 'water'
diccionario['gato'] = 'cat'
diccionario['raton'] = 'mouse'
diccionario['aire'] = 'air'
diccionario['arbol'] = 'tree'
diccionario['puerta'] = 'door'
diccionario['chica'] = 'girl'
diccionario['ojo'] = 'eye'
diccionario['zorro'] = 'fox'
diccionario['tigre'] = 'tiger'
diccionario['lobo'] = 'wolf'
diccionario['tiburon'] = 'shark'
diccionario['maiz'] = 'corn'
diccionario['oso'] = 'bear'
diccionario['perro'] = 'dog'
diccionario['serpiente'] = 'snake'
diccionario['cerdo'] = 'pig'
diccionario['opcion'] = 'option'
diccionario['palabra'] = 'word'
diccionario['coche'] = 'car'

print(diccionario)

'''
paso todas las claves del diccioanrio a una lista
'''
for x, y in diccionario.items():
    lista += [x]
# print(lista)

'''
añadimos la primera palabra
'''
aux = lista[randint(0, 19)]
lista5.append(aux)
# print(lista5)

'''
comparamos palabras para que no se repitan
'''
for i in range(0, 4):
    while lista5.__contains__(aux):
        aux = lista[randint(0, 19)]
    lista5.append(aux)
  
print(lista5)

'''
titulo del programa
'''
print("\tAver si aciertas!")
print("\t=================")
print()

for i in range(0, 5):
    palabraEs = lista5[i]
    print("Introduzca la traduccion de ", palabraEs, ": ")
    palabraIn = input()
    if palabraIn.__eq__(diccionario.get(palabraEs)):
        cont1 = cont1 + 1
    else:
        cont2 = cont2 + 1

print("Tu numero de aciertos es de: ", cont1 , " \nTu numero de fallos es de: ", cont2)

File: coleccionesDiccionarios/Ejercicio11.java
/**Realiza un programa que escoja al azar 5 palabras en español del minidiccionario
del ejercicio anterior. El programa irá pidiendo que el usuario
teclee la traducción al inglés de cada una de las palabras y comprobará si
son correctas. Al final, el programa deberá mostrar cuántas respuestas son
válidas y cuántas erróneas.

@author Rafael Infante
*/
package coleccionesDiccionarios;

import java.util.HashMap;
import java.util.Scanner;
import java.util.ArrayList;

public class Ejercicio11 {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);

		// variables
		int i;
		String aux = "";
		String palabraEs = "";
		String palabraIn = "";
		int cont1 = 0; // contador aciertos
		int cont2 = 0; // contador fallos

		// creacion del mapa
		HashMap<String, String> diccionario = new HashMap<String, String>();

		// creo un arraylist "copia del mapa"
		ArrayList<String> diccionarioArray = new ArrayList<String>();

		// creo otro Arraylist donde almacenar 5 palabras
		ArrayList<String> diccionarioCinco = new ArrayList<String>();

		// palabras con su traduccion
		diccionario.put("gato", "cat");
		diccionario.put("raton", "mouse");
		diccionario.put("agua", "water");
		diccionario.put("perro", "dog");
		diccionario.put("arbol", "tree");
		diccionario.put("edad", "age");
		diccionario.put("aire", "air");
		diccionario.put("arte", "art");
		diccionario.put("centimo", "cent");
		diccionario.put("ciudad", "city");
		diccionario.put("maiz", "corn");
		diccionario.put("dia", "day");
		diccionario.put("puerta", "door");
		diccionario.put("ojo", "eye");
		diccionario.put("chica", "girl");

		// En este arraylist almacenare todas las claves del mapa
		diccionarioArray.addAll(diccionario.keySet());

		// añadimos la primera palabra
		aux = diccionarioArray.get((int) (Math.random() * 15));
		diccionarioCinco.add(aux);

		// Aqui comparamos para que no se nos repita
		for (i = 0; i < 4; i++) {

			do {
				aux = diccionarioArray.get((int) (Math.random() * 15));
			} while (diccionarioCinco.contains(aux));
			diccionarioCinco.add(aux);
		}

		// titulo del programa
		System.out.println("\tAver si aciertas!");
		System.out.println("\t=================");
		System.out.println();

		for (i = 0; i < 5; i++) {
			palabraEs = diccionarioCinco.get(i);
			System.out.println("Introduzca la traduccion de " + palabraEs + ": ");
			palabraIn = sc.nextLine();
			if (palabraIn.equals(diccionario.get(palabraEs))) {
				cont1++;
			} else {
				cont2++;
			}
		}

		System.out.println("Tu numero de aciertos es de: " + cont1 + " \nTu numero de fallos es de: " + cont2);

	}

}
